Lib/Src/CUartCom.cpp

```diff
@@ -98,25 +100,40 @@ void CUartCom::stopRx()
 }
 
 /**
- * @brief Add data to the TX Queue and start transmission if possible
- * @param msg Message to send.
- * @return True if message was added to the queue and transmission started
- * successfully.
+ * @brief send string
+ * @param msg Message to be sent via UART
+ * @return success if message was successfully stored in TX Queue.
  */
 bool CUartCom::send(etl::string<MAX_STRING_SIZE> msg)
 {
-    bool b_success = false;
-    // Add message to queue
-    if (m_tx_queue.size() <= MAX_TX_QUEUE_SIZE && (msg.empty() == false))
+    uint32_t msg_len = msg.length();
+    bool b_success = send((uint8_t *)msg.c_str(), msg_len);
+    return b_success;
+}
+
+/**
+ * @brief Add data to the TX Queue and start transmission if possible
+ * @param p_data_buf Reference to array of uint8_t data
+ * @param len Size of data that needs to be stored
+ * @return True if data was added to the queue, false in case of buffer overflow
+ */
+bool CUartCom::send(uint8_t *p_data_buf, uint32_t len)
+{
+    bool b_success = true;
+
+    // Store data in tx_queue
+    for (uint32_t i = 0; i < len; i++)
     {
-        m_tx_queue.put(msg);
-        b_success = true;
+        if (m_tx_queue.put(p_data_buf[i]) == false)
+        {
+            b_success = false;
+        }
     }
 
-    // Start transmission only if UART is idle
+    // Start transmission only if UART is not transmitting already
```

Comment:
This is really nit picking, but the comment does not match the code. Your code says:
**If idle then transmit**, while your comment says **If not transmitting then transmit**. It is not clear to the code reader that these states are a complete set of possible states. Your code is correct in state transitions. You should go into transmit state only from Idle state, so changing the comment is the best approach.

Reply:
Or I'd remove the comment altogether. It does not add to comprehension. Your code is very well written and does not require any aid to understand.